test/integration_test.exs

```diff
@@ -3,42 +3,79 @@ defmodule AvalancheIntegrationTest do
 
   alias Avalanche.TestHelper
 
-  @tag integration: true
+  @moduletag integration: true
+
   describe "run/2 with OAuth token" do
     setup do
       options = TestHelper.test_options()
       [options: options]
     end
 
-    test "returns a Response struct", c do
-      assert {:ok, %Avalanche.Response{} = response} = Avalanche.run("select 1;", [], c.options)
-      assert response.status == 200
+    test "returns a Result struct", c do
+      assert {:ok, %Avalanche.Result{} = result} = Avalanche.run("select 1;", [], c.options)
+      assert result.num_rows == 1
     end
   end
 
-  @tag integration: true
   describe "run/2 with Key Pair token" do
     setup do
       options = TestHelper.test_key_pair_options()
       [options: options]
     end
 
-    test "returns a Response struct", c do
-      assert {:ok, %Avalanche.Response{} = response} = Avalanche.run("select 1;", [], c.options)
-      assert response.status == 200
+    test "returns a Result struct", c do
+      assert {:ok, %Avalanche.Result{} = result} = Avalanche.run("select 1;", [], c.options)
+      assert result.num_rows == 1
     end
   end
 
-  @tag integration: true
   describe "run/2" do
     setup do
       options = TestHelper.test_options()
       [options: options]
     end
 
     test "allows bind variables", c do
-      assert {:ok, %Avalanche.Response{} = response} = Avalanche.run("select ?;", [33], c.options)
-      assert response.status == 200
+      assert {:ok, %Avalanche.Result{} = result} = Avalanche.run("select ?;", [33], c.options)
+      assert result.num_rows == 1
+    end
+
+    test "parses result body into list of maps", c do
+      assert {:ok, %Avalanche.Result{} = result1} =
+               Avalanche.run(
+                 "SELECT *, 9 as number FROM SNOWFLAKE_SAMPLE_DATA.TPCH_SF1.ORDERS LIMIT ?",
```

Comment:
should we have a sort on this query to ensure that the row that you are looking for is the first one returned?

Reply:
good call.